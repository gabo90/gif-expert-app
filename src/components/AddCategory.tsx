import { useState } from "react"

interface AddCategoryProps {
    onNewCategory: Function
}

export const AddCategory = ({onNewCategory}:AddCategoryProps) => {
    const [inputValue, setInputValue] = useState('')
    
    const onInputChange = ({target}:any) => {
        setInputValue(target.value);
    }

    const onSubmit = (event:any) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}