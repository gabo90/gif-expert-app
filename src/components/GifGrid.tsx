import useFetchGifs from '../hooks/useFetchGifs';
import { GridItem } from '.';

interface GifGridProps {
    category: String
}

export const GifGrid = ({category}:GifGridProps) => {
    
    const {images, isLoading} = useFetchGifs(category)
    
    return (
        <>
            <h2>{category}</h2>

            {
                isLoading && (<h2>Cargando...</h2>)
            }
            <div className="card-grid">
                {
                    images.map((image:any) => (
                        <GridItem
                            key={image.id}
                            {...image}
                        />         
                    ))
                }
            </div>
        </>
    )
}