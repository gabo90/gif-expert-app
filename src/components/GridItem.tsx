interface GridItemProps {
    title: string
    url: string
}

export const GridItem = ({title, url}: GridItemProps) => {

    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}