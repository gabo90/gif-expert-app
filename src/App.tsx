import { useState } from "react"
import { AddCategory, GifGrid } from './components';


const App = () => {

    const [categories, setcategories] = useState(['Marvel']);

    const onAddCategory = (newCategory:any) => {
        setcategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>Gift Expert App</h1>

            <AddCategory onNewCategory={onAddCategory}/>

            {
                categories.map((category) => (
                        <GifGrid
                            key={category}
                            category={category}
                        />
                    )
                )
            }
        </>
    )
}

export default App