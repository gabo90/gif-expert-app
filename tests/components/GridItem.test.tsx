import { render, screen } from "@testing-library/react";
import { GridItem } from '../../src/components';

describe('Pruebas en <GridItem/>', () => {

    const title = 'Marvel';
    const url = "https://marvel.com/image.jpg";

    test('debe hacer match con el snapshot', () => {
        const {container} = render(<GridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe contener la url y el alt correspondiente', () => {
        render(<GridItem title={title} url={url} />);
        
        const {src, alt}: HTMLImageElement = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe mostrar el titulo en el componente', () => {
        render(<GridItem title={title} url={url} />);
        
        expect(screen.getByText(title)).toBeTruthy();
    });
});