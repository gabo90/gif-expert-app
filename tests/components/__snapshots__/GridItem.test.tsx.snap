// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GridItem/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Marvel"
      src="https://marvel.com/image.jpg"
    />
    <p>
      Marvel
    </p>
  </div>
</div>
`;
